File: Tyche.Backend/Felipearpa.Tyche.Pool/Felipearpa.Tyche.Pool/Domain/PoolGamblerScoreMapper.fs
namespace Felipearpa.Tyche.Pool.Domain

open Felipearpa.Type

module PoolGamblerScoreMapper =

    let mapToDomain (poolGamblerScoreEntity: PoolGamblerScoreEntity) =
        { PoolGamblerScore.PoolId = poolGamblerScoreEntity.PoolId |> Ulid.newOf
          PoolName = poolGamblerScoreEntity.PoolName |> NonEmptyString100.newOf
          GamblerId = poolGamblerScoreEntity.GamblerId |> Ulid.newOf
          GamblerUsername = poolGamblerScoreEntity.GamblerUsername |> NonEmptyString100.newOf
          CurrentPosition = poolGamblerScoreEntity.CurrentPosition |> Option.ofNullable
          BeforePosition = poolGamblerScoreEntity.BeforePosition |> Option.ofNullable
          Score = poolGamblerScoreEntity.Score |> Option.ofNullable }

File: Tyche.Android/pool/src/main/java/com/felipearpa/tyche/pool/ui/PoolGamblerScoreMapper.kt
package com.felipearpa.tyche.pool.ui

import com.felipearpa.tyche.pool.domain.PoolGamblerScore

fun PoolGamblerScore.toModel() =
    PoolGamblerScoreModel(
        poolId = this.poolId,
        poolName = this.poolName,
        gamblerId = this.gamblerId,
        gamblerUsername = this.gamblerUsername,
        currentPosition = this.currentPosition,
        beforePosition = this.beforePosition,
        score = this.score
    )